### PR TITLE
Bug 1184869 - Support new MozMobileMessage.getSmscAddress impl.

### PR DESCRIPTION
**Pull request by freesamael**
Update SIMSlot.getSmsc() to support both legacy and new
MozMobileMessage.getSmscAddress() implementations.

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a718af2b1847cb4529d0cf2c282edc09b35bae81

File: shared/js/simslot.js [hunk 1]
```diff
@@ -138,16 +138,24 @@
...
-        console.error(this.error);
-        cb(null);
-      };
+      // The return type of getSmscAddress can be either a DOMRequest which
+      // resolves to a string for legacy implementations, or
+      // a Promise which resolves to a SmscAddress object in Bug 1043250.
+      mobileMessage.getSmscAddress(this.index).then(
+        (result) => {
+          var smsc;
+          if (typeof result === 'string' || result instanceof String) {
```
Comment (evelynhung) [inline shared/js/simslot.js]: nit: add a comment to note it's for backward compatible.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=de9bbee81f91ffeaea2e6f24cdf8496e5f10bfbf

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=369d349c5753135b4bc29ac990caec344dda16d6

**Pull request merged**